quinn/src/connection.rs

```diff
@@ -143,194 +152,25 @@ impl Connecting {
     ///
     /// On all non-supported platforms the local IP address will not be available,
     /// and the method will return `None`.
-    ///
-    /// Will panic if called after `poll` has returned `Ready`.
     pub fn local_ip(&self) -> Option<IpAddr> {
-        match *self.0.as_ref().unwrap() {
-            ConnectingState::Incoming(ref incoming) => incoming.inner.local_ip(),
-            ConnectingState::Handshaking(ref handshaking) => handshaking.local_ip(),
-            ConnectingState::AcceptError(_) => panic!("local_ip called after connecting errored"),
-        }
+        let conn = self.conn.as_ref().unwrap();
+        let inner = conn.state.lock("local_ip");
+
+        inner.inner.local_ip()
     }
 
-    /// The peer's UDP address
+    /// The peer's UDP address.
```

Comment:
nit: no trailing punctuation on the first line

Reply:
Presumably you were looking at solely the newer commits when reviewing--I reverted all changes to this file, and so the reason the period was re-added is because it is already like that on main. Regardless, I can drive-by fix it.

Reply:
Oops, right.